backend/logic/courses/courses.js

```diff
@@ -401,6 +401,50 @@ var assignCourseToCurriculum = async (courseType, courseCode, curriculum) => {
   return true;
 };
 
+/**
+ * Saves user preferences to the database.
+ * @author Steven Li
+ * @param {int} student_id A student id
+ * @param {JSON} courses An array of course codes
+ * @throws invalid format course code
+ *         invalid format student id
+ *         mysql connection errors
+ */
+var saveUserPreferences = async (student_id, courses) => {
+  if (!format.verifyStudentId(student_id)) {
+    throw new Error("invalid format student id");
+  }
+  if (!courses) {
+    throw new Error("invalid format course code");
```

Comment:
Rather than "invalid format course code", wouldn't that be instead "empty courses list" or "no courses provided"?

Reply:
Fixed! and added an extra test case for it